connection.go

```diff
@@ -1094,13 +1128,31 @@ func (conn *Connection) send(req Request, streamId uint64) *Future {
 		default:
 		}
 	}
+
+	if atomic.LoadUint32(&(conn.shutdownInProgress)) == shutdownInProgress {
+		conn.cancelFuture(fut, fmt.Errorf("server shutdown in progress"))
+		return fut
+	}
```

Comment:
What happens if we send a request during a graceful shutdown? Now the code does not protect against this.

Reply:
> What happens if we send a request during a graceful shutdown?
It's either will be processed or will be declined, depending on is our graceful shutdown has been started.

Reply:
AFAIU, no.

> When a client is notified about the upcoming server exit, it stops serving any new requests and waits for active requests to complete before closing the connections.

so each request that squeeze through before we started our shutdown (set `conn.shutdownInProgress`) considered to be one of the "active requests" we execute.

Reply:
I'll try to test this one.

Reply:
I'm confusing about:

https://github.com/tarantool/tarantool/blob/master/src/box/lua/net_box.c#L196-L197

Reply:
Yeah, it seems that Tarantool doesn't decline any requests we send after it started shutdown process, it executes them like everything is still ok. So it's up to us to decide what requests is "active requests".

Reply:
Of course, it's in our interest to minimize the quantity of active requests.

Reply:
Then it's easier to make the check before a request execution (in Connection.Do()).

Reply:
After state machine rework I check "is shutdown now" in the same place where we check that connection is not closed/disconnected. I think it would be less confusing to have them in the same place.